ui/src/store/types.ts

```diff
@@ -1,13 +1,40 @@
 import { VuexOidcState } from 'vuex-oidc'
+import { DatasetCore, Quad, Term } from 'rdf-js'
 import { AppState } from './modules/app'
 import { APIState } from './modules/api'
 import { ProjectsState } from '@/store/modules/projects'
 import { ProjectState } from '@/store/modules/project'
+import { Actions } from '@/api/mixins/ApiResource'
+import { SharedDimensionsState } from './modules/sharedDimensions'
+import { SharedDimensionState } from './modules/sharedDimension'
+import { GraphPointer } from 'clownface'
+import { ResourceIdentifier } from 'alcaeus'
+import { TypeCollection } from '@tpluscode/rdfine/lib/TypeCollection'
 
 export interface RootState {
   app: AppState
   auth: VuexOidcState
   api: APIState
   projects: ProjectsState
   project: ProjectState
+  sharedDimensions: SharedDimensionsState
+  sharedDimension: SharedDimensionState
+}
+
+export interface Resource {
+  id: ResourceIdentifier
+  clientPath: string
+  actions: Actions
+  pointer: GraphPointer
+  types: TypeCollection<DatasetCore<Quad, Quad>>
+}
+
+export interface SharedDimension extends Resource {
+  name?: string
+  terms?: Term
+}
+
+export interface SharedDimensionTerm extends Resource {
+  name: Term[]
+  identifiers: string[]
```

Comment:
Does the `Resource` interface essentially duplicate `import { RdfResource } from '@tpluscode/rdfine'` and could be substituted?

Reply:
Kinda. It's technically a subset of `RdfResource`. My types are a bit broken because I'm using the types from `@cube-creator/model` when they exist, but my serializers don't really respect them. This `Resource` is actually more correct.